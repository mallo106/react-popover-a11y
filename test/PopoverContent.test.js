import React, { createRef } from 'react'
import { mount, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Container, { PopoverContent } from '../source/PopoverContent'

describe('PopoverContent', () => {

  describe('container', () => {

    it('renders', () => {
      const ref = createRef()
      const wrapper = mount(
        <div>
          <Container
            id="mockId"
            label="Mock label"
            ref={ref}
            triggerRect={{
              left: 300,
              height: 18,
              top: 200,
              width: 18
            }}
          >
            <div>This is some content</div>
          </Container>
        </div>
      )
      const tree = toJson(wrapper)
      expect(tree).toMatchSnapshot()
    })

  })

  describe('component', () => {

    it('renders', () => {
      const focus = jest.fn()
      const forwardedRef = { current: { focus } }
      const wrapper = shallow(
        <PopoverContent
          forwardedRef={forwardedRef}
          id="mockId"
          label="Mock label"
          triggerRect={{
            left: 0,
            height: 0,
            top: 0,
            width: 0
          }}
        >
          <div>This is some content</div>
        </PopoverContent>
      )
      const tree = toJson(wrapper)
      expect(tree).toMatchSnapshot()
    })

    it('renders, triggerRect has values', () => {
      const focus = jest.fn()
      const forwardedRef = { current: { focus } }
      const wrapper = shallow(
        <PopoverContent
          forwardedRef={forwardedRef}
          id="mockId"
          label="Mock label"
          triggerRect={{
            left: 300,
            height: 18,
            top: 200,
            width: 18
          }}
        >
          <div>This is some content</div>
        </PopoverContent>
      )
      const tree = toJson(wrapper)
      expect(tree).toMatchSnapshot()
    })

    it('renders, custom style provided', () => {
      const focus = jest.fn()
      const forwardedRef = { current: { focus } }
      const wrapper = shallow(
        <PopoverContent
          forwardedRef={forwardedRef}
          id="mockId"
          label="Mock label"
          triggerRect={{
            left: 300,
            height: 18,
            top: 200,
            width: 18
          }}
          style={{ zIndex: '9999' }}
        >
          <div>This is some content</div>
        </PopoverContent>
      )
      const tree = toJson(wrapper)
      expect(tree).toMatchSnapshot()
    })

    it('renders, additional props added', () => {
      const focus = jest.fn()
      const forwardedRef = { current: { focus } }
      const wrapper = shallow(
        <PopoverContent
          forwardedRef={forwardedRef}
          id="mockId"
          label="Mock label"
          triggerRect={{
            left: 300,
            height: 18,
            top: 200,
            width: 18
          }}
          data-foo="bar"
          onClick={jest.fn()}
        >
          <div>This is some content</div>
        </PopoverContent>
      )
      const tree = toJson(wrapper)
      expect(tree).toMatchSnapshot()
    })

    it('renders, triggerRect null', () => {
      const focus = jest.fn()
      const forwardedRef = { current: { focus } }
      const wrapper = shallow(
        <PopoverContent
          forwardedRef={forwardedRef}
          id="mockId"
          label="Mock label"
          triggerRect={null}
          style={{ zIndex: '9999' }}
        >
          <div>This is some content</div>
        </PopoverContent>
      )
      const tree = toJson(wrapper)
      expect(tree).toMatchSnapshot()
    })

    it('renders, full mount', () => {
      const focus = jest.fn()
      const forwardedRef = { current: { focus } }
      const wrapper = mount(
        <PopoverContent
          forwardedRef={forwardedRef}
          id="mockId"
          label="Mock label"
          triggerRect={{
            left: 300,
            height: 18,
            top: 200,
            width: 18
          }}
        >
          <div>This is some content</div>
        </PopoverContent>
      )
      const tree = toJson(wrapper)
      expect(tree).toMatchSnapshot()
    })

    it('calls focus on forwardedRef on mount', done => {
      const forwardedRef = createRef()
      const wrapper = mount(
        <PopoverContent
          forwardedRef={forwardedRef}
          id="mockId"
          label="Mock label"
          triggerRect={{
            left: 300,
            height: 18,
            top: 200,
            width: 18
          }}
        >
          <div>This is some content</div>
        </PopoverContent>
      )

      const focusSpy = jest.spyOn(forwardedRef.current, 'focus')

      wrapper.instance().componentDidMount()

      setTimeout(() => {
        expect(focusSpy).toHaveBeenCalled()
        done()
      }, 0)
    })

  })

})
