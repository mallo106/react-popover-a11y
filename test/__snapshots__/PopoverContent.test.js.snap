// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopoverContent component renders 1`] = `
<div
  aria-label="Mock label"
  id="mockId"
  role="dialog"
  style={
    Object {
      "left": 0,
      "position": "fixed",
      "top": 0,
      "zIndex": "999",
    }
  }
  tabIndex="-1"
>
  <div>
    This is some content
  </div>
</div>
`;

exports[`PopoverContent component renders, additional props added 1`] = `
<div
  aria-label="Mock label"
  data-foo="bar"
  id="mockId"
  onClick={[MockFunction]}
  role="dialog"
  style={
    Object {
      "left": 313.5,
      "position": "fixed",
      "top": 213.5,
      "zIndex": "999",
    }
  }
  tabIndex="-1"
>
  <div>
    This is some content
  </div>
</div>
`;

exports[`PopoverContent component renders, custom style provided 1`] = `
<div
  aria-label="Mock label"
  id="mockId"
  role="dialog"
  style={
    Object {
      "left": 313.5,
      "position": "fixed",
      "top": 213.5,
      "zIndex": "9999",
    }
  }
  tabIndex="-1"
>
  <div>
    This is some content
  </div>
</div>
`;

exports[`PopoverContent component renders, full mount 1`] = `
<PopoverContent
  forwardedRef={
    Object {
      "current": <div
        aria-label="Mock label"
        id="mockId"
        role="dialog"
        style="z-index: 999; left: 313.5px; position: fixed; top: 213.5px;"
        tabindex="-1"
      >
        <div>
          This is some content
        </div>
      </div>,
    }
  }
  id="mockId"
  label="Mock label"
  triggerRect={
    Object {
      "height": 18,
      "left": 300,
      "top": 200,
      "width": 18,
    }
  }
>
  <div
    aria-label="Mock label"
    id="mockId"
    role="dialog"
    style={
      Object {
        "left": 313.5,
        "position": "fixed",
        "top": 213.5,
        "zIndex": "999",
      }
    }
    tabIndex="-1"
  >
    <div>
      This is some content
    </div>
  </div>
</PopoverContent>
`;

exports[`PopoverContent component renders, triggerRect has values 1`] = `
<div
  aria-label="Mock label"
  id="mockId"
  role="dialog"
  style={
    Object {
      "left": 313.5,
      "position": "fixed",
      "top": 213.5,
      "zIndex": "999",
    }
  }
  tabIndex="-1"
>
  <div>
    This is some content
  </div>
</div>
`;

exports[`PopoverContent component renders, triggerRect null 1`] = `
<div
  aria-label="Mock label"
  id="mockId"
  role="dialog"
  style={
    Object {
      "left": 0,
      "position": "fixed",
      "top": 0,
      "zIndex": "9999",
    }
  }
  tabIndex="-1"
>
  <div>
    This is some content
  </div>
</div>
`;

exports[`PopoverContent container renders 1`] = `
<div>
  <ForwardRef(withForwardedRef(PopoverContent))
    id="mockId"
    label="Mock label"
    triggerRect={
      Object {
        "height": 18,
        "left": 300,
        "top": 200,
        "width": 18,
      }
    }
  >
    <PopoverContent
      forwardedRef={
        Object {
          "current": <div
            aria-label="Mock label"
            id="mockId"
            role="dialog"
            style="z-index: 999; left: 313.5px; position: fixed; top: 213.5px;"
            tabindex="-1"
          >
            <div>
              This is some content
            </div>
          </div>,
        }
      }
      id="mockId"
      label="Mock label"
      triggerRect={
        Object {
          "height": 18,
          "left": 300,
          "top": 200,
          "width": 18,
        }
      }
    >
      <div
        aria-label="Mock label"
        id="mockId"
        role="dialog"
        style={
          Object {
            "left": 313.5,
            "position": "fixed",
            "top": 213.5,
            "zIndex": "999",
          }
        }
        tabIndex="-1"
      >
        <div>
          This is some content
        </div>
      </div>
    </PopoverContent>
  </ForwardRef(withForwardedRef(PopoverContent))>
</div>
`;
